### PR TITLE
RUBY-1126 Support collation

### PR DESCRIPTION
**Pull request by estolfo**

### CODE REVIEW

File: lib/mongo/bulk_write.rb [hunk 1]
```diff
@@ -182,11 +186,18 @@ def insert_one(documents, server, operation_id)
     end
 
     def update(documents, server, operation_id)
```
Comment (jyemin) [inline lib/mongo/bulk_write.rb]: collation should be applied to delete as well as update

File: lib/mongo/collection/view/builder/find_command.rb [hunk 2]
```diff
@@ -94,6 +94,7 @@ def specification
 
           def find_command
             document = BSON::Document.new('find' => collection.name, 'filter' => filter)
+            document[:collation] = collation if collation
```
Comment (jyemin) [inline lib/mongo/collection/view/builder/find_command.rb]: Is validate_collation called at a higher level?

File: lib/mongo/bulk_write.rb (hunk 1 above)
Comment (estolfo) [inline lib/mongo/bulk_write.rb]: Yes, I'm not quite done with this PR; it's a WIP. 
I've just commit BulkWrite delete support: https://github.com/mongodb/mongo-ruby-driver/pull/788/commits/6c94dbbe8e1b0fd91034f45967ad7aa6b407c97b

File: lib/mongo/collection/view/builder/find_command.rb (hunk 2 above)
Comment (estolfo) [inline lib/mongo/collection/view/builder/find_command.rb]: yes, it's called here: https://github.com/mongodb/mongo-ruby-driver/pull/788/files/3321ddd3889c17d49a534502e137e44d16404e76#diff-cf306a3295c965d0c497491ba4d63ea6R64
before the FindCommand Builder even has a chance to build the command.

File: lib/mongo/bulk_write.rb (hunk 1 above)
Comment (jyemin) [inline lib/mongo/bulk_write.rb]: Acknowledged. Thanks for the clarification.

File: lib/mongo/collection/view/builder/find_command.rb (hunk 2 above)
Comment (jyemin) [inline lib/mongo/collection/view/builder/find_command.rb]: Acknowledged

**Comment by estolfo** [comment]
@rozza
Can you take a look at this? Thank you!

**Comment by rozza** [comment]
LGTM!

Its been an age since I did any ruby so somethings took some deciphering like `feature_enabled?` especially in the crud_spec - very neat.  Also `tap` is new to me and is really useful.

**Comment by estolfo** [comment]
thanks @rozza = )

**Pull request merged**